src/GitHubAuthController.php

```diff
@@ -56,10 +58,22 @@ protected function getAuthorizationUrlOptions()
     /**
      * {@inheritdoc}
      */
-    protected function getIdentification(ResourceOwnerInterface $resourceOwner)
+    protected function getIdentification(ResourceOwnerInterface $resourceOwner, AbstractProvider $provider, AccessToken $token)
     {
+        $email = $resourceOwner->getEmail();
+        if (!$email) {
+            $url = $provider->getResourceOwnerDetailsUrl($token).'/emails';
+            $emails = $provider->getResponse($provider->getAuthenticatedRequest('GET', $url, $token));
+            foreach ($emails as $e) {
```

Comment:
I have a strong feeling this code can be cleaned up a little. Also please try to refrain from using such short variables, especially in this case `$e` is often also used for storing an `Exception`.

Maybe split the whole follow up request into a seperate method?